analysis.py

```diff
@@ -1,52 +1,61 @@
-# This analysis retrieves the device list of your account and print to the console.
-# There are examples on how to apply filter.
-#
-# Environment Variables
-# In order to use this analysis, you must setup the Environment Variable table.
-#
-# account_token: Your account token
-#
-# Steps to generate an account_token:
-# 1 - Enter the following link: https://admin.tago.io/account/
-# 2 - Select your Profile.
-# 3 - Enter Tokens tab.
-# 4 - Generate a new Token with Expires Never.
-# 5 - Press the Copy Button and place at the Environment Variables tab of this analysis.
-
-from tago import Account
-from tago import Analysis
-
-# The function myAnalysis will run when you execute your analysis
-def myAnalysis(context, scope):
-  # reads the value of account_token from the environment variable
-  account_token = list(filter(lambda account_token: account_token['key'] == 'account_token', context.environment))
-  account_token = account_token[0]['value']
-
-  if not account_token:
-    return context.log("Missing account_token Environment Variable.")
-  
-  my_account = Account(account_token)
-
-  # Example of filtering devies by Tag.
-  # You can filter by: name, last_input, last_output, bucket, etc.
-  my_filter = {
-    'tags': [{
-      'key': 'keyOfTagWeWantToSearch', value: 'valueOfTagWeWantToSearch',
-    }],
-    # 'bucket': '55d269211a2e236c25bb9859',
-    # 'name': 'My Device'
-    # 'name': 'My Dev*
-  }
-  # Searching all devices with tag we want
-  devices = account.devices.list(1, ['id', 'tags'], my_filter, 10000)
-  if devices['status'] is True:
-    context.log(devices['result']) # Array with data
-  else:
-    context.log(devices['message']) # Error (if status is False)
+"""
+ Analysis Example
+ Get Device List
+
+ This analysis retrieves the device list of your account and print to the console.
+ There are examples on how to apply filter.
+
+ Environment Variables
+ In order to use this analysis, you must setup the Environment Variable table.
+
+ account_token: Your account token
+
+ Steps to generate an account_token:
+ 1 - Enter the following link: https://admin.tago.io/account/
+ 2 - Select your Profile.
+ 3 - Enter Tokens tab.
+ 4 - Generate a new Token with Expires Never.
+ 5 - Press the Copy Button and place at the Environment Variables tab of this analysis.
+"""
+from tagoio_sdk import Account, Analysis
+from tagoio_sdk.modules.Account.Device_Type import DeviceInfoList
+
+
+def query_devices(account: Account) ->list[DeviceInfoList]:
+	# Example of filtering devices by Tag.
+	# You can filter by: name, last_input, last_output, bucket, etc.
+	my_filter = {
+		"tags": [
+			{"key": "keyOfTagWeWantToSearch", "value": "valueOfTagWeWantToSearch"}
+		],
+		# "bucket": "55d269211a2e236c25bb9859",
+		# "name": "My Device"
+	}
+
+	devices = account.devices.listDevice({
+		"page": 1,
+		"fields": ["id", "tags"],
+		"filter": my_filter,
+		"amount": 20
+	})
+
+	return devices
+
+
+def list_devices(context: list[dict], scope: list) -> None:
```

Comment:
Create a issue to add types to the analysis context in the future.

setattr(context, "log", log)
setattr(context, "token", token)
setattr(context, "environment", environment)
setattr(context, "analysis_id", analysis_id)